deepmd_utils/model_format/output_def.py

```diff
@@ -0,0 +1,278 @@
+# SPDX-License-Identifier: LGPL-3.0-or-later
+from typing import (
+    Dict,
+    List,
+    Tuple,
+    Union,
+)
+
+
+def check_var(var, var_def):
+    if var_def.atomic:
+        # var.shape == [nf, nloc, *var_def.shape]
+        if len(var.shape) != len(var_def.shape) + 2:
+            raise ValueError(f"{var.shape[2:]} length not matching def {var_def.shape}")
+        if list(var.shape[2:]) != var_def.shape:
+            raise ValueError(f"{var.shape[2:]} not matching def {var_def.shape}")
+    else:
+        # var.shape == [nf, *var_def.shape]
+        if len(var.shape) != len(var_def.shape) + 1:
+            raise ValueError(f"{var.shape[1:]} length not matching def {var_def.shape}")
+        if list(var.shape[1:]) != var_def.shape:
+            raise ValueError(f"{var.shape[1:]} not matching def {var_def.shape}")
```

Comment:
**Suggestion:** The `check_var` function is used in multiple places in the code. It would be more efficient to refactor it into a class method of `VariableDef` or `OutputVariableDef`. This way, you can directly call it on an instance of these classes, making the code more readable and maintainable. [maintainability]
```suggestion
class VariableDef:
    ...
    def check_var(self, var):
        if self.atomic:
            # var.shape == [nf, nloc, *self.shape]
            if len(var.shape) != len(self.shape) + 2:
                raise ValueError(f"{var.shape[2:]} length not matching def {self.shape}")
            if list(var.shape[2:]) != self.shape:
                raise ValueError(f"{var.shape[2:]} not matching def {self.shape}")
        else:
            # var.shape == [nf, *self.shape]
            if len(var.shape) != len(self.shape) + 1:
                raise ValueError(f"{var.shape[1:]} length not matching def {self.shape}")
            if list(var.shape[1:]) != self.shape:
                raise ValueError(f"{var.shape[1:]} not matching def {self.shape}")
```